### PR TITLE
Add gateway monitor metrics and event

### LINKED ISSUE #2233: More metrics and events about gateway status
Metrics:
- [ ] Gateway packet count/size
- [ ] Gateway packet retransmission

Events:
- [ ] Subnet gateway type changes
- [ ] Subnet gateway node changes
- [ ] Gateway status changes to not ready

### PR DESCRIPTION
**Pull request by changluyi**
- [ ] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

#### What type of this PR
Examples of user facing changes:
- Features
    1. add gateway packet count metrics
    2. add event for subnet gatewayType or gatewayNode change
    3. add event for subnet gateway ping failed
- Bug fixes
- Docs
- Tests

#### Which issue(s) this PR fixes:
Fixes #2233

### CODE REVIEW

File: pkg/daemon/metrics.go [hunk 1]
```diff
@@ -66,6 +66,28 @@ var (
 		[]string{"code", "method", "host"},
 	)
 
+	metricOvnSubnetGatewayPacketBytes = prometheus.NewGaugeVec(prometheus.GaugeOpts{
```
Comment (oilbeater) [inline pkg/daemon/metrics.go:69]: We should use counter type as the number keeps increase

File: pkg/daemon/controller_linux.go [hunk 2]
```diff
@@ -531,6 +531,11 @@ func (c *Controller) loopEncapIpCheck() {
 	}
 }
 
+func (c *Controller) ovnMetricsUpdate() {
+	resetOvnSubnetGatewayMetrics()
```
Comment (oilbeater) [inline pkg/daemon/controller_linux.go:535]: why reset is needed

**Review by oilbeater — CHANGES_REQUESTED** [review]

File: pkg/daemon/gateway_linux.go [hunk 3]
```diff
@@ -571,6 +571,48 @@ func (c *Controller) setIptables() error {
...
+		var ovnSubnetGatewayCountRules []util.IPTableRule
+		_, subnetCidrs, err := c.getDefaultVpcSubnetsCIDR(protocol)
+		if err != nil {
+			klog.Errorf("get subnets failed, %+v", err)
+			return err
+		}
+
+		for name, subnetCidr := range subnetCidrs {
+			ovnSubnetGatewayCountRules = []util.IPTableRule{
+				{Table: "filter", Chain: "FORWARD", Rule: strings.Fields(fmt.Sprintf(`-m comment --comment %s,%s -s %s`, util.OvnSubnetGatewayIptables, name, subnetCidr))},
```
Comment (oilbeater) [inline pkg/daemon/gateway_linux.go:583]: Should skip the traffic to the host, which will not be sent to external network

File: pkg/daemon/gateway_linux.go [hunk 4]
```diff
@@ -571,6 +571,48 @@ func (c *Controller) setIptables() error {
...
+				{Table: "filter", Chain: "FORWARD", Rule: strings.Fields(fmt.Sprintf(`-m comment --comment %s,%s -d %s`, util.OvnSubnetGatewayIptables, name, subnetCidr))},
```
Comment (oilbeater) [inline pkg/daemon/gateway_linux.go:584]: The `cleanup.sh` script should remove related iptable rules

**Review by oilbeater — CHANGES_REQUESTED** [review]

File: pkg/daemon/gateway_linux.go (hunk 3 above)
Comment (changluyi) [reply pkg/daemon/gateway_linux.go:583]: FORWARD chain do not calculate the packet to host ip. it is calculate on INPUT chain

File: pkg/daemon/gateway_linux.go (hunk 4 above)
Comment (changluyi) [reply pkg/daemon/gateway_linux.go:584]: if the subnet is removed, this rule will be removed. its releases in 
image

**Review by oilbeater — APPROVED** [review]

**Comment by oilbeater** [comment]
The subnet metrics relies on these rules, we should consider a batch way to update these rules.

**Pull request merged**